src/pal/src/configure.cmake

```diff
@@ -42,10 +42,7 @@ if(NOT CMAKE_SYSTEM_NAME STREQUAL Darwin)
   check_include_files("libintl.h" HAVE_LIBINTL_H)
 endif()
 
-if(NOT CMAKE_SYSTEM_NAME STREQUAL FreeBSD AND NOT CMAKE_SYSTEM_NAME STREQUAL NetBSD)
-  set(CMAKE_REQUIRED_FLAGS "-ldl")
-endif()
-check_include_files(lttng/tracepoint.h HAVE_LTTNG_TRACEPOINT_H)
+find_path (HAVE_LTTNG_TRACEPOINT_H NAMES lttng/tracepoint.h)
```

Comment:
cmake produces this linker command: `/usr/bin/gcc -ldl /path/to/file.o -o output` for `check_include_files`. GCC requires `-ldl` to be the last option: `/usr/bin/gcc /path/to/file.o -o output -ldl`, while clang options are order-independent.

I couldn't figure out another way to make cmake 3.10x to work with `check_include_files`, where `-ldl` is required by the header introspection.  However, with cmake 3.12x and above, we can use: `CMAKE_REQUIRED_LIBRARIES ${CMAKE_DL_LIBS}`. `find_path` works for the entire matrix.

Reply:
I have thought about this change and I think that we should not use the find_path. The problem with that one is that cmake looks for the path, so I am worried that it may happen that it finds it in a different location than the compiler, especially for cross building.

Can you please replace it with
```cmake
check_cxx_source_compiles("
#include <lttng/tracepoint.h>
int main(int argc, char **argv) {
  return 0;
}" HAVE_LTTNG_TRACEPOINT_H)
```

Reply:
Ah, your `check_cxx_source_compiles` approach surely is more robust. I will replace it with that. Thanks.

Btw, I noticed that `find_type` is also used in CoreCLR for `UTYPES_H` and `LLDB` headers detection. Perhaps those two places could get similar upgrades at some point?

Reply:
These cases are different - we want to look at the specific locations that we pass to the find_path in those cases.

Reply:
Changed to `check_cxx_source_compiles` with conditional `set(CMAKE_REQUIRED_LIBRARIES ${CMAKE_DL_LIBS})` and unset. Both clang and gcc seem satisfied.